src/topology/config/vars.rs

```diff
@@ -2,15 +2,17 @@ use regex::{Captures, Regex};
 use std::collections::HashMap;
 
 pub fn interpolate(input: &str, vars: &HashMap<String, String>) -> String {
-    let re = Regex::new(r"\$\$|\$(\w+)|\$\{(\w+)\}").unwrap();
+    let re = Regex::new(r"\$\$|\$(\w+)|\$\{(\w+)(?::-([^}]+)?)?\}").unwrap();
```

Comment:
If I'm reading this right, the `([^}]+)?` section is the default value and we allow it to be empty. What happens in that case (e.g. `${FOO:-}`)?

Reply:
> If I'm reading this right

Yeah, the regex is getting a bit unreadable, I guess it's right at the cusp of splitting it up into newlines with some documentation to explain what it does, but I figured this was still OK.

> What happens in that case (e.g. `${FOO:-}`)?

An empty value is/should be an empty string (which can be a useful default value), but that's definitely a test that was still missing. I've just added it.

Technically, the default when a variable can't be found is already to set it to an empty string, but it has the side-effect of triggering a warning in the logs, which it wouldn't in this case.

Reply:
> Yeah, the regex is getting a bit unreadable, I guess it's right at the cusp of splitting it up into newlines with some documentation to explain what it does, but I figured this was still OK.

Yep, I agree this is still perfectly fine. There will likely be a point where this kind of function is absorbed into the normal config file parsing but we're not quite there yet.

> An empty value is converted to an empty string (which can be a useful default value), but that's definitely a test that was still missing. I've just added it.
> 
> Technically, when a variable is unset its value is already set to an empty string, but that has the side-effect of triggering a warning in the logs, which doesn't happen in this format.

That makes sense to me, thanks!